src/wallet/rpcwallet.cpp

```diff
@@ -2575,6 +2577,234 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
+    if (!SignalsOptInRBF(wtx))
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Transaction is not bip-125 replaceable");
+
+    // Check that if the original tx has wallet conflicts (meaning there is at least one other tx
+    // in the wallet that spends at least one of the same outputs) that it is the most recent one.
+    // This prevents us from trying to bump fee on a tx that has already been bumped.  So if the
+    // user created txid1, bumped the fee which resulted in txid2, and now wishes to bump the fee
+    // again, the user must call bumpfee on txid2.
+    const CWalletTx *conflictTx;
+    BOOST_FOREACH(const uint256& conflict, wtx.GetConflicts()) {
+        conflictTx = pwalletMain->GetWalletTx(conflict);
+        LogPrint("rpc", "bumpfee wallet conflict, conflicted tx has time: %d\n", conflictTx->GetTxTime());
+        if (conflictTx->GetTxTime() > wtx.GetTxTime()) {
+            strError = strprintf("Transaction conflicts with later wallet tx: %s", conflict.GetHex());
+            throw JSONRPCError(RPC_INVALID_PARAMETER, strError);
+        }
+    }
+
+    // bounds-check the output to decrement
+    int nOutput = request.params[1].get_int();
+    if (nOutput < 0 || (unsigned int) nOutput > wtx.vout.size())
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Output out of bounds");
+
+    // optional parameters
+    int newConfirmTarget = nTxConfirmTarget;
+    CAmount totalFee = 0;
+    if (request.params.size() > 2) {
+        RPCTypeCheck(request.params, boost::assign::list_of(UniValue::VSTR)(UniValue::VNUM)(UniValue::VOBJ));
+        UniValue options = request.params[2];
+        RPCTypeCheckObj(options,
+            {
+                {"confTarget", UniValueType(UniValue::VNUM)},
+                {"totalFee", UniValueType(UniValue::VNUM)},
+            },
+            true, true);
+
+        if (options.exists("confTarget")) {
+            newConfirmTarget = options["confTarget"].get_int();
+            if (newConfirmTarget <= 0) // upper-bound will be checked by estimatefee/smartfee
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid confTarget (cannot be <= 0)");
+        }
+        if (options.exists("totalFee")) {
+            totalFee = options["totalFee"].get_int();
+            if (totalFee <= 0)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid totalFee (cannot be <= 0)");
+            else if (totalFee > maxTxFee)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid totalFee (higher than maxTxFee)");
+        }
+    }
+
+    // calculate the old fee and fee-rate
+    CAmount nDebit = wtx.GetDebit(ISMINE_SPENDABLE);
+    CAmount nOldFee = (wtx.IsFromMe(ISMINE_SPENDABLE) ? nDebit - wtx.GetValueOut() : 0);
+    int txSize = (int)GetVirtualTransactionSize((CTransaction)wtx);
+    CFeeRate nOldFeeRate(nOldFee, txSize);
+
+    // use the user-defined payTxFee if possible, otherwise use smartfee / fallbackfee
+    CAmount nNewFee = 0;
```

Comment:
Using a nNewFee = 0 is fine here, it just slightly bothers me to use this as a null value when indeed 0 is a valid fee amount.

-1 would be an invalid fee amount (creates coins) so I have a slight preference to use this.